frontend/packages/ceph-storage-plugin/src/types.ts

```diff
@@ -248,3 +248,41 @@ export type NavUtils = {
   getIndex: (searchSpace: any, search: string, offset?: number) => number;
   getAnchor: (step: number, mode: number) => string;
 };
+export type StoreAction =
+  | { type: 'setSecretName'; value: string }
+  | { type: 'setSecretKey'; value: string }
+  | { type: 'setAccessKey'; value: string }
+  | { type: 'setRegion'; value: string }
+  | { type: 'setTarget'; value: string }
+  | { type: 'setEndpoint'; value: string };
+
+export type ProviderDataState = {
+  secretName: string;
+  secretKey: string;
+  accessKey: string;
+  region: string;
+  target: string;
+  endpoint: string;
+};
+
+export type Payload = K8sResourceCommon & {
+  spec: {
+    type: string;
+    ssl: boolean;
+    [key: string]: any;
+  };
+};
+
+export type BackingStoreProviderDataState = ProviderDataState & {
+  numVolumes: number;
+  gcpJSON: string;
+  volumeSize: string;
+  storageClass: string;
+};
+
+export type BackingStoreAction =
+  | StoreAction
+  | { type: 'setGcpJSON'; value: string }
+  | { type: 'setVolumes'; value: number }
+  | { type: 'setVolumeSize'; value: string }
+  | { type: 'setStorageClass'; value: string };
```

Comment:
we create `reducer.ts` file to store actions and state related to a store.
See `/ocs-install/internal-mode/`

Reply:
done